### PR TITLE
feat(federation): Link from Federated API to Origin Integration

### PR DESCRIPTION
**Pull request by pes-spyro-soft-com**
## Issue

https://gravitee.atlassian.net/browse/APIM-4948
https://gravitee.atlassian.net/browse/APIM-4917
https://gravitee.atlassian.net/browse/APIM-4978

## Description

- Link from a Federated API to the Integration it was created from
- Connection wizard for Solace, Refactor existing form 
- Information message when discover in process

## Additional context

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-fcijeuzplj.chromatic.com)

### CODE REVIEW

File: gravitee-apim-console-webui/src/management/api/general-info/api-general-info.component.ts [hunk 1]
```diff
@@ -99,6 +104,7 @@ export class ApiGeneralInfoComponent implements OnInit, OnDestroy {
     private readonly permissionService: GioPermissionService,
     private readonly snackBarService: SnackBarService,
     private readonly matDialog: MatDialog,
+    private integrationsService: IntegrationsService,
```
Comment (ThibaudAV) [inline gravitee-apim-console-webui/src/management/api/general-info/api-general-info.component.ts]: What do you think of keeping the same convention as everywhere in the console code and keeping the services with `readonly` ?

File: gravitee-apim-console-webui/src/management/api/general-info/api-general-info.component.ts [hunk 2]
```diff
@@ -111,7 +117,7 @@ export class ApiGeneralInfoComponent implements OnInit, OnDestroy {
 
     combineLatest([this.apiService.get(this.apiId), this.categoryService.list()])
       .pipe(
-        switchMap(([api, categories]) =>
+        switchMap(([api, categories]: [Api, Category[]]) =>
```
Comment (ThibaudAV) [inline gravitee-apim-console-webui/src/management/api/general-info/api-general-info.component.ts]: Normally, the inferred type should be enough, no? if it's not good, there's probably a problem at the source of the type. 🤔

File: gravitee-apim-console-webui/src/management/integrations/integration-agent/integration-agent.configuration.ts [hunk 3]
```diff
@@ -31,3 +31,27 @@ services:
...
+    environment:
+      - gravitee_integration_connector_ws_endpoints_0=\${WS_ENDPOINTS}
+      - gravitee_integration_connector_ws_headers_0_name=Authorization
+      - gravitee_integration_connector_ws_headers_0_value=bearer \${WS_AUTH_TOKEN}
+      - gravitee_integration_providers_0_configuration_authToken=\${SOLACE_AUTH_TOKEN}
+      - gravitee_integration_providers_0_configuration_0_appDomains=\${SOLACE_APPLICATION_0_DOMAIN:-}
+      - gravitee_integration_providers_0_configuration_1_appDomains=\${SOLACE_APPLICATION_1_DOMAIN:-}
+      - gravitee_integration_providers_0_integrationId=\${INTEGRATION_ID}
+      - gravitee_integration_providers_0_type=solace
+      - gravitee_integration_providers_0_configuration_url=\${SOLACE_ENDPOINT:-https://apim-production-api.solace.cloud/api/v2/apim}
```
Comment (michel-barret) [inline gravitee-apim-console-webui/src/management/integrations/integration-agent/integration-agent.configuration.ts]: I suggest

- little simplification for non helpful without our `.env`
- put at end the optionals configurations

```suggestion
      - gravitee_integration_connector_ws_endpoints_0=\${WS_ENDPOINTS}
      - gravitee_integration_connector_ws_headers_0_name=Authorization
      - gravitee_integration_connector_ws_headers_0_value=bearer \${WS_AUTH_TOKEN}
      - gravitee_integration_providers_0_configuration_authToken=\${SOLACE_AUTH_TOKEN}
      - gravitee_integration_providers_0_integrationId=\${INTEGRATION_ID}
      - gravitee_integration_providers_0_type=solace
      - gravitee_integration_providers_0_configuration_url=https://apim-production-api.solace.cloud/api/v2/apim
      # optionals
      - gravitee_integration_providers_0_configuration_0_appDomains=\${SOLACE_APPLICATION_0_DOMAIN}
      - gravitee_integration_providers_0_configuration_1_appDomains=\${SOLACE_APPLICATION_1_DOMAIN}
```

**Review by ThibaudAV — COMMENTED** [review]
LGTM but I leave to the team to make the final ✅  😁

File: gravitee-apim-console-webui/src/management/api/general-info/api-general-info.component.ts (hunk 2 above)
Comment (pes-spyro-soft-com) [reply gravitee-apim-console-webui/src/management/api/general-info/api-general-info.component.ts]: yeah, it inferred the type correctly. To me, the advantage for explicit typing type is easier navigation through the code as the type annotations work like hyperlinks, also we are more explicit what we are expecting.

Comment (ThibaudAV) [reply gravitee-apim-console-webui/src/management/api/general-info/api-general-info.component.ts]: The problem is that sometimes this forces the type, which means that if the upstream type changes, the typing doesn't inform us that we've changed the contract. 
by example here if you add `[ApiV2, Category[]]` no error and yet it's not an ApiV2. 

It's up to your IDE to provide you with the means to navigate inferred types, but not to be added to the code IMHO

**Review by michel-barret — APPROVED** [review]

**Pull request merged**